### PR TITLE
Add homeassistant.reload_all service

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Documents the new `homeassistant.reload_all` service.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/87769
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/homeassistant.markdown [hunk 1]
```diff
@@ -23,6 +23,17 @@ The `homeassistant` integration provides services for controlling Home Assistant
...
 
+### Service `homeassistant.reload_all`
+
+Reload all YAML configuration that can be reloaded without restarting Home Assistant.
+
+It calls the `reload` service on all domains that have it available. Additionally,
+it reloads the core configuration (equivalent to calling
+`homeassistant.reload_core_config`) and themes (`frontend.reload_themes`).
+
+Prior to reloading, it run a basic configuration check. If that fails, the reload
```
Comment (teharris1) [inline source/_integrations/homeassistant.markdown]: Change to "Prior to reloading, a basic configuration check is performed."

Comment (frenck) [reply source/_integrations/homeassistant.markdown]:
```suggestion
Prior to reloading, a basic configuration check is performed. If that fails, the reload
```

Comment (frenck) [reply source/_integrations/homeassistant.markdown]: Thanks!

**Review by emontnemery — APPROVED** [review]
Thanks @frenck 👍

**Review by piitaya — APPROVED** [review]

**Pull request merged**